Dockerfile

```diff
@@ -1,6 +1,16 @@
 FROM registry.svc.ci.openshift.org/openshift/release:golang-1.12 AS builder
 WORKDIR /go/src/github.com/openshift/cluster-logging-operator
-COPY . .
+
+# COPY steps are in the reverse order of frequency of change
+COPY cmd ./cmd
+COPY version ./version
+COPY scripts ./scripts
+COPY files ./files
+COPY manifests ./manifests
+COPY vendor ./vendor
+COPY Makefile ./Makefile
+COPY pkg ./pkg
```

Comment:
Can we regroup these according to dependency and frequency order please? For example cmd, version, scripts and files should not affect vendor. You check my sample.

Reply:
we can either have relative order based on frequency of change OR based on dependency, not both. I have ordered in based on decreasing order of frequency of change.

My intention behind this PR is to replace `COPY . . ` with more reasonable set of `COPY`, without impacting correctness, nothing else.  Relative order of COPY affects usage of cache, affects efficiency not correctness as far as i understand. Thats why this order of `COPY` statements. `cmd, version, scripts, files` is less likely to change, so these are at top. `pkg` is most like to change, so it is at bottom. `vendor` is more likely to change than `cmd, version, scripts, files` so somewhere in middle.

Reply:
Sorry for the misunderstanding, I fully understand your intention and would also see this `COPY . .` vanish sooner than later. 

My statement is deps order first so that each group is depending on each other for correctness and then within each group frequency order. IMHO if something in vendor changes everything in `cmd/version/pkg` should be copied anew. For example, a bump of the operator-sdk in vendor affects immediately `cmd/manager/main.go`.

Reply:
if `vendor` changes, and there is a corresponding change in `cmd/pkg/version` wouldnt that be detected by buildah? and if there is no change, it need not be copied again.
As far as i can see, there is no impact on correctness w.r.t. the order of COPY. correctness is impacted if we miss to copy something totally.
If it is so sensitive on order, i would suggest we go back to `COPY . . ` then we have taken care of all conditions.